File: NodeJS/selectors/select_by_role.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto('https://example.com');
  const heading = await page.getByRole('heading');
  console.log('By Role:', await heading.textContent());
  await browser.close();
})();

File: Python/selectors/select_by_role.py
from playwright.sync_api import sync_playwright

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        page.goto("https://example.com")
        heading = page.get_by_role("heading")
        print("By Role:", heading.inner_text())
        browser.close()

run()